lib/services/UserServices.js

```diff
@@ -0,0 +1,41 @@
+const User = require('../models/User');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+module.exports = class UserService {
+  static async create({ email, password }) {
+    const passwordHash = await bcrypt.hash(password, Number(process.env.SALT_ROUNDS));
+    const user = await User.insert({ email, passwordHash });
+    return user;
+  }
+  
+  static async authorize({ email, password }) {
+    try {
+      const user = await User.findByEmail(email);
+      const passwordsMatch = await bcrypt.compare(password, user.passwordHash);
+      if(!passwordsMatch) throw new Error('Invalid Password');
+      return user;
+    } catch(err) {
+      err.status = 401;
+      throw err;
+    }
+  }
+
+  // update admin password by email via put route
+  static async changePassword(userId, password) {
+    const newPasswordHash = await bcrypt.hash(password, Number(process.env.SALT_ROUNDS));
+    const user = await User.findById(userId);
+    return User.update({ ...user, password: newPasswordHash });
+  }
+
+  static authToken(user) {
+    return jwt.sign({ user }, process.env.APP_SECRET, {
+      expiresIn: '24h'
+    });
+  }
+
+  static verifyAuthToken(token) {
+    const { user } = jwt.verify(token, process.env.APP_SECRET);
+    return user;
+  }
+};
```

Comment:
The authToken() function on Line 31 looked slightly different in my group's Tardygram lab. I'm not sure what the difference really means or how it matters, but I'll share it in case you can tell the difference or if it matters. The only difference was that the other group added this to the user on Line 32 --> user: user.toJSON()

static authToken(user) {
    return jwt.sign(
      { user: user.toJSON() }, 
      process.env.APP_SECRET,
      { expiresIn: '24h' }
    );
  }